spoken/views.py

```diff
@@ -77,7 +78,12 @@ def home(request):
             status=2), tutorial_detail__foss_id=f, language__name='English').order_by('tutorial_detail__order')[:1].first()
         random_tutorials.append((tcount, tutorial))
     try:
-        tr_rec = TutorialResource.objects.filter(Q(status=1) | Q(status=2)).order_by('?')[:1].first()
+        tr_rec = None
+        queryset = TutorialResource.objects.filter(Q(status=1) | Q(status=2))
+        count = queryset.count()
+        if count > 0:
+            random_index = randint(0, count - 1)
+            tr_rec = queryset.all()[random_index]
```

Comment:
```suggestion
            tr_rec = queryset[random_index]
```
this avoids the all()